ops/testing.py

```diff
@@ -2295,50 +2423,55 @@ def __delitem__(self, key: str) -> None:
     def __len__(self):
         return len(self._children)
 
-    def create_dir(self, name: str, **kwargs) -> '_Directory':
-        self._children[name] = _Directory(self.path / name, **kwargs)
-        return self._children[name]
+    def create_dir(self, name: str, **kwargs: Any) -> '_Directory':
+        dirc = _Directory(self.path / name, **kwargs)
+        self._children[name] = dirc
+        return dirc
 
     def create_file(
             self,
             name: str,
-            data: typing.Union[bytes, str, typing.BinaryIO, typing.TextIO],
-            encoding: typing.Optional[str] = 'utf-8',
-            **kwargs
+            data: 'ReadableBuffer',
+            encoding: Optional[str] = 'utf-8',
+            **kwargs: Any
     ) -> '_File':
-        self._children[name] = _File(self.path / name, data, encoding=encoding, **kwargs)
-        return self._children[name]
+        file = _File(self.path / name, data, encoding=encoding, **kwargs)
+        self._children[name] = file
+        return file
 
 
 class _File:
     def __init__(
             self,
             path: pathlib.PurePosixPath,
-            data: typing.Union[str, bytes, typing.BinaryIO, typing.TextIO],
-            encoding: typing.Optional[str] = 'utf-8',
-            **kwargs):
+            data: 'ReadableBuffer',
```

Comment:
I do like the simplicity of some of those file and buffer related objects that you've created. The code is a lot easier to understand, this way.